ADO.NET/ResultSetPrinter.cs

```diff
@@ -2,65 +2,103 @@
-        foreach (DbColumn col in columns)
+        
+        private void printColumnHeaders() 
         {
-            int columnWidth = getPrintedColumnWidth (col);
-            Console.Write("{0} ", col.ColumnName.PadRight (columnWidth));
+            printColumnSeparator();
+
+            // Write the column headers
+            foreach (DbColumn column in m_columns) {
+                int columnWidth = getPrintedColumnWidth(column);
+                checkColumnWidth(columnWidth);
+                Console.Write("{0} ", column.ColumnName.PadRight(columnWidth));
+            }
+            Console.WriteLine();
+
+            printColumnSeparator();
         }
-        Console.WriteLine();
 
-        // Write a column separator
-        foreach (DbColumn col in columns)
+        private string formatColumnValue(DbColumn column, string Value)
         {
-            int columnWidth = getPrintedColumnWidth (col);
-            Console.Write("{0} ", pad ('-', columnWidth));
+            checkColumnWidth(Value.Length);
+            int columnWidth = getPrintedColumnWidth(column);
+            if(Value.Length > columnWidth) {
+                m_ValuesTruncated = true;
+                // Truncate value if its too long
+                return Value.Substring(0, columnWidth - TRUNCATION_SYMBOL.Length) + TRUNCATION_SYMBOL;
+            }
+            return Value.PadRight(columnWidth);
         }
-        Console.WriteLine();
-    }
 
-    private int getPrintedColumnWidth (DbColumn column)
-    {
-        int width = column.ColumnName.Length > column.ColumnSize ? column.ColumnName.Length : (int)column.ColumnSize;
-        if (maxColumnWidth != -1) {
-            if (width > maxColumnWidth) return maxColumnWidth;
+        private void printColumnSeparator() {
+            // Write a column separator
+            foreach (DbColumn col in m_columns) {
+                int columnWidth = getPrintedColumnWidth(col);
+                Console.Write("{0} ", pad('-', columnWidth));
+            }
+            Console.WriteLine();
         }
 
-        return width;
-    }
+        private void checkColumnWidth(int columnWidth) 
+        {
+            if (m_widestColumnRead < columnWidth) {
+                m_widestColumnRead = columnWidth;  
+            }          
+        }
 
-    private string pad (char value, int length)
-    {
-        string paddedString = "".PadRight (length);
-        if (value == ' ')
+        private int getPrintedColumnWidth(DbColumn column)
+        { 
+            int columnMinWidth = Math.Max(column.ColumnName.Length, TRUNCATION_SYMBOL.Length);
+            if(m_maxColumnWidth < 0) {
+                return Math.Max(columnMinWidth, (int)column.ColumnSize);
+            }
+            return Math.Max(columnMinWidth, m_maxColumnWidth);  
+        }
+
+        private string pad(char value, int length)
         {
-            return paddedString;
+            string paddedString = "".PadRight(length);
```

Comment:
I think it's probably better to use the string constructor that handles this case insetead of PadRight, don't you think?
i.e.
string paddedString = new string(' ', length)